File: LeetCode Solutions/March 2022 LeetCoding Challenge/Validate Stack Sequences/Validate Stack Sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        int lenP = pushed.size(), i = 0, j = 0, sp = 0;
        while(i < lenP){
            if(sp >= 0 && popped[j] == pushed[sp]){ 
                j++;
                sp--;
            } else{
                sp++;
                i++;
                if(i < lenP){
                    pushed[sp] = pushed[i];
                }
            }
        }
        return sp == 0;
    }
};

File: LeetCode Solutions/March 2022 LeetCoding Challenge/Validate Stack Sequences/Validate Stack Sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        lenP, i, j, sp = len(pushed), 0, 0, 0
        while(i < lenP):
            if(~sp and popped[j] == pushed[sp]):
                j += 1
                sp -= 1
            else:
                sp += 1
                i += 1
                if(i < lenP):
                    pushed[sp] = pushed[i]
        return not sp